src/app/model/ChessEngine.ts

```diff
@@ -119,4 +136,31 @@ export class ChessEngine implements ChessModel {
   public possibleMoves(position: Square): Square[] {
     return this.pieceMovesNotCausingAllyKingCheckmate(position);
   }
+
+  private kingWasChecked(chosenPiece: Piece): KingWasChecked | undefined {
+    const kingsSide = chosenPiece.side === Side.WHITE ? Side.BLACK : Side.WHITE;
+    const kingPosition = this.kingPosition(this.board, kingsSide);
+    // poniższe brzydkie i mi się nie podoba, ale na razie działa - możecie podpowiedzieć zmiany :)
+    let king;
+    if (kingPosition) {
+      king = this.board.onPositionPiece(kingPosition);
+    }
+    const isKingChecked = this.isKingChecked(this.board, kingsSide);
+    return isKingChecked && king instanceof King && kingPosition
+      ? {
+          eventType: 'KingWasChecked',
+          king: king,
+          onSquare: kingPosition,
+        }
+      : undefined;
+  }
+
+  private kingWasUnchecked(pieceWasMoved: PieceWasMoved): KingWasUnchecked | undefined {
+    const king = pieceWasMoved.piece;
+    const kingLastPosition = pieceWasMoved.from;
+
+    return king instanceof King ? { eventType: 'KingWasUnchecked', king: king, onSquare: kingLastPosition } : undefined;
+  }
 }
+
+export type ChessboardEvent = PieceWasMoved | PieceWasCaptured | KingWasChecked | KingWasUnchecked;
```

Comment:
Fajnie, ze to wyciągnąłeś, też miałem to w końcu zaproponować :) Pomyślałbym tylko nad nazwa, bo będą jeszcze inne eventy (do promocji). Np. 
zrobić taki typ? 
`type MoveResult = (PieceWasMoved | PieceWasCaptured | KingWasChecked | KingWasUnchecked)[]`

Reply:
no właśnie nad ta nazwą się zastanawiałem, ale że było późno to zostawiłem tak ;)

Reply:
Tzn. to by było OK i mi się podoba :) Ale już wybiegam w przód.